src/utils/retry.js

```diff
@@ -0,0 +1,25 @@
+const retry = async (
+  asyncRequest,
+  onInitialError,
+  totalAttempts = 10,
+  sleepTime = 2000
+) => {
+  let lastError;
```

Comment:
found this a little misleading, can you rename it to something like `latestError` or `previousError`?